iceoryx_posh/include/iceoryx_posh/experimental/internal/popo/base_publisher.inl

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_BASE_PUBLISHER_INL
+#define IOX_EXPERIMENTAL_POSH_POPO_BASE_PUBLISHER_INL
+
+namespace iox {
+namespace popo {
+
+using uid_t = uint64_t;
+
+template<typename T, typename port_t>
+BasePublisher<T, port_t>::BasePublisher(const capro::ServiceDescription&)
+    /* : m_port(iox::runtime::PoshRuntime::getInstance().getMiddlewareSender(service, "")) */
+{}
+
+template<typename T, typename port_t>
+inline uid_t
+BasePublisher<T, port_t>::uid() const noexcept
+{
+    std::cout << "uid()" << std::endl;
+    return 0u;
+}
+
+template<typename T, typename port_t>
+inline cxx::expected<Sample<T>, AllocationError>
+BasePublisher<T, port_t>::loan(uint32_t size) noexcept
+{
+    auto result = m_port.tryAllocateChunk(size);
+    if(result.has_error())
+    {
+        return cxx::error<AllocationError>(result.get_error());
+    }
+    else
+    {
+        return cxx::success<Sample<T>>(convertChunkHeaderToSample(result.get_value()));
+    }
+}
+
+template<typename T, typename port_t>
+inline void
+BasePublisher<T, port_t>::publish(Sample<T>& sample) noexcept
+{
+    if(!isOffered())
```

Comment:
nice try. we shouldn't do this. In ara::com you must be able to update the value of a field prior to offering it. just forward the sample to the next layer here. This will take care

Reply:
(/)